### PR TITLE
Add Font Awesome 6 to devupdate (blocked by MDL-76989)

### PR DESCRIPTION
**Pull request by sarjona**
A reference to the devupdate page for Moodle 4.2 should be added to mention Font Awesome has been upgraded from 4.7 to 6.3.0 in [MDL-76989](https://tracker.moodle.org/browse/MDL-76989).

<a href="https://gitpod.io/#https://github.com/moodle/devdocs/pull/569"></a>

### CODE REVIEW

File: docs/devupdate.md
Comment (andrewnicols) [inline docs/devupdate.md:123]: minor grammar: 

```suggestion
Font Awesome 6 is backwards compatible (because a shim has been included too), so the old syntax still works.
```

File: docs/devupdate.md
Comment (andrewnicols) [inline docs/devupdate.md:125]:
```suggestion
More information about the changes between 4 and 6, including any icons which have been renamed, can be found in https://fontawesome.com/docs/web/setup/upgrade/upgrade-from-v4
```

**Review by andrewnicols — CHANGES_REQUESTED** [review]

**Comment by sarjona** [comment]
Hi @andrewnicols!
Thanks a lot for reviewing the patch and suggesting these improvements in the wording! <3 I appreciate it a lot! :-*

**Comment by andrewnicols** [comment]
Holding off on merge ubtil 76989 lands.

As a side-note, it would be super if we could add a GHA check of some kind to have it block until the issue lands, at which point it auto merges.

No: CI Lint is failing here.

**Comment by sarjona** [comment]
Thanks Andrew!

The lint errors are unrelated to this page. I've created a PR to fix them: https://github.com/moodle/devdocs/pull/571

I love the idea of the GHA job to block issues until they land (although for now, I think that's something that can be done manually by integrators when we merge the issue).

**Comment by sarjona** [comment]
I've added more information to explain some more changes that developers might need to apply to their code.

**Comment by sarjona** [comment]
Now that [MDL-76989](https://tracker.moodle.org/browse/MDL-76989) has been integrated, it would be great having this merged too :-)

**Comment by sarjona** [comment]
@andrewnicols could you take a look at the changes I added last week (after you reviewed it) :-)

**Pull request merged**
